tests/FileStorageAbsTest.php

```diff
@@ -0,0 +1,138 @@
+                $this->fail('List of Azure Blob Storages contains also S3 Storage');
+            }
+        }
+    }
+
+    public function testSetAbsStorageAsDefault()
+    {
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->assertFalse($storage['isDefault']);
+
+        $storage = $this->client->setAbsFileStorageAsDefault($storage['id']);
+
+        $this->assertTrue($storage['isDefault']);
+
+        $storageList = $this->client->listAbsFileStorage();
+        foreach ($storageList as $item) {
+            if ($item['isDefault'] && $item['id'] !== $storage['id']) {
+                $this->fail('There are more storage with default flag');
+            }
+        }
+    }
+
+    public function testCrossProviderStorageDefaultAbsS3()
+    {
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('AWS S3 file storage "%d" not found', $storage['id']));
+        $this->client->setS3FileStorageAsDefault($storage['id']);
+    }
+
+    public function testCreateAbsStorageWithoutRequiredParam()
+    {
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage('"accountName": This field is missing. "accountKey": This field is missing. "containerName": This field is missing. "owner": This field is missing.');
+        $this->client->createAbsFileStorage([]);
+    }
+
+    public function testRotateAbsCredentialsWithoutRequiredParams()
+    {
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage('"accountKey": This field is missing');
+
+        $this->client->rotateAbsFileStorageCredentials($storage['id'], []);
+    }
+
+    public function testProjectAssignAbsFileStorage()
+    {
+        $name = 'My org';
+        $organization = $this->client->createOrganization($this->testMaintainerId, [
+            'name' => $name,
+        ]);
+
+        $project = $this->client->createProject($organization['id'], [
+            'name' => 'My test',
+        ]);
+
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->client->assignFileStorage($project['id'], $storage['id']);
+
+        $project = $this->client->getProject($project['id']);
+        $this->assertEquals($storage['id'], $project['fileStorage']['id']);
+    }
+
+    public function testCrossProviderStorageCredentialsRotateAbsS3()
+    {
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('AWS S3 file storage "%d" not found', $storage['id']));
+        $this->client->rotateS3FileStorageCredentials($storage['id'], self::ROTATE_S3_OPTIONS);
+    }
+}
```

Comment:
koukám, že tady lítaj newliny na konci souborů

Reply:
Jo. Už má rozdělaný PR na zavedení PHPCS https://github.com/keboola/kbc-manage-api-php-client/pull/79, kde jsem změnil asi 32 souborů. To bude review za odměnu :smile: 

Tady zatím dodělám NL.